src/app/app.component.ts

```diff
@@ -265,7 +412,84 @@ export class AppComponent implements AfterViewInit {
   onEntrySelected(event) {
     const customEvent = event as CustomEvent<LfTreeNode[]>;
     const treeNodesSelected: LfTreeNode[] = customEvent.detail;
-    this.entrySelected = treeNodesSelected?.length > 0 ? treeNodesSelected[0] : undefined;
+    this.entrySelected =
+      treeNodesSelected?.length > 0 ? treeNodesSelected[0] : undefined;
+  }
+
+  async onToolbarOptionSelected(event) {
+    let customEvent = event as CustomEvent;
+    await customEvent.detail.tag.handler();
+  }
+
+  openNewFolderDialog(): void {
+    this.dialog.open(NewFolderModalComponent, {
+      data: { makeNewFolder: this.makeNewFolder.bind(this) },
+    });
+  }
+
+  openEditColumnsDialog(): void {
+    this.dialog.open(EditColumnsModalComponent, {
+      data: {
+        columnsSelected: this.selectedColumns,
+        allColumnOptions: this.allPossibleColumns,
+        updateColumns: this.updateColumns.bind(this),
+      },
+    });
+  }
+
+  async makeNewFolder(folderName: string) {
+    if (folderName) {
+      if (!this.lfRepositoryBrowser?.nativeElement?.currentFolder) {
+        throw new Error(
+          this.localizationService.getString('NO_CURRENTLY_OPENED_FOLDER')
+        );
+      }
+      await this.addNewFolderAsync(
+        this.lfRepositoryBrowser?.nativeElement?.currentFolder,
+        folderName
+      );
+      await this.lfRepositoryBrowser?.nativeElement?.refreshAsync();
+    } else {
+      throw new Error(
+        this.localizationService.getString('PLEASE_PROVIDE_FOLDER_NAME')
+      );
+    }
+  }
+
+  async addNewFolderAsync(
+    parentNode: LfTreeNode,
+    folderName: string
+  ): Promise<void> {
+    if (!this.repoClient) {
+      throw new Error('repoClient is undefined');
+    }
+    type RequestParameters = {
+      entryId: number;
+      postEntryChildrenRequest: PostEntryChildrenRequest;
+    };
+
+    const entryId =
+      (parentNode as LfRepoTreeNode).targetId ?? parseInt(parentNode.id, 10);
+    const postEntryChildrenRequest: PostEntryChildrenRequest =
+      new PostEntryChildrenRequest({
+        name: folderName,
+        entryType: PostEntryChildrenEntryType.Folder,
+      });
+    const requestParameters: RequestParameters = {
+      entryId,
+      postEntryChildrenRequest,
+    };
+    const repoId: string = await this.repoClient.getCurrentRepoId();
+    await this.repoClient?.entriesClient.createOrCopyEntry({
+      repoId,
+      entryId: requestParameters.entryId,
```

Comment:
is there a reason you created the type and object? If not, you can just assign the variables directly here